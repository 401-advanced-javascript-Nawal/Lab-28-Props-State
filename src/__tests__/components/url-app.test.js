import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import URLapp from '../../components/url-app.js';


describe ('<URL-App />' , () => {

  it(' It Should be exists in our applications ', () => {
    // virtual 
    let app = shallow(<URLapp />);
    // find the tag which already includes in our app 
    expect(app.find('h1').exists()).toBeFalsy();
  }); // end of it test 

  it(' Renders correctly ' , () => {
    const tree = renderer.create(<URLapp />).toJSON();
    expect(tree).toMatchSnapshot();
  }); // end of it test for render 

}); // end of describe test 