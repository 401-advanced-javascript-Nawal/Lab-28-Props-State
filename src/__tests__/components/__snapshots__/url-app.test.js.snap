// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<URL-App />  Renders correctly  1`] = `
Array [
  <h1>
     URL Full Address : 
    
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <section>
      <label
        className="apiC"
      >
         API URL :
        <input
          onChange={[Function]}
          type="text"
        />
      </label>
    </section>
    <section
      className="methodC"
    >
      <button
        className="methods"
        onClick={[Function]}
        value="get"
      >
         GET
      </button>
      <button
        className="methods"
        onClick={[Function]}
        value="post"
      >
         POST
      </button>
      <button
        className="methods"
        onClick={[Function]}
        value="put"
      >
         PUT
      </button>
      <button
        className="methods"
        onClick={[Function]}
        value="patch"
      >
         PATCH
      </button>
      <button
        className="methods"
        onClick={[Function]}
        value="delete"
      >
         DELETE
      </button>
      <button
        className="submitB"
      >
         GO!
      </button>
    </section>
    <p>
       The Results 
    </p>
    <textarea
      type="text"
    />
  </form>,
]
`;
